Creating Post Section

alaasaedy opened the pull request:
- creating Post Section with its storybook and passing necessary props.
- using ReusableDiv component to display post images also inner icons.
- using PopUp component inside image as option button.
- use grid to align images in a row with a gap of 16 px.
- using of document width to set images width in small screens.
- click function added to the pickIcon to add background to it.

alaasaedy committed:
Creating PostSection component

alaasaedy committed:
saving current change

alaasaedy committed:
creating post section

alaasaedy committed:
remove comments

alaasaedy committed:
- add Like, save, share bar
- change shareBtn width, height

alaasaedy committed:
adding no repeat, and change background size to 100% instead of cover and setting background position to top center.
wrap every reusable div with a div to control its width on different screen sizes.
remove unnecessary styling to set responsive.

adding bottom votes, save, share bar

MAlngaawy commented on .vscode/settings.json:
```diff
@@ -0,0 +1,4 @@
+{
+    "terminal.integrated.fontFamily": "Consolas, 'Courier New', monospace",
+    "editor.fontLigatures": false,
+}
```
please remove this file from the PR

MAlngaawy commented on src/components/DivWithCenterdChildren/index.js:
```diff
@@ -32,7 +32,9 @@ export const ReusableDiv = ({
-        backgroundSize: 'cover',
+        backgroundRepeat: 'no-repeat',
+        backgroundPosition: 'center top',
+        backgroundSize: '100%',
```
don't change the props here.
if you need a prop just add it, and make it conditionally.
but don't delete or replace any prop from here.

alaasaedy replied:
those aren't a props, I'm changing the div background image size to fit our uses because make it cover makes me not able to scale it in different screen sizes.

MAlngaawy commented on src/components/PostSection/index.jsx:
```diff
@@ -0,0 +1,117 @@
+  searchIcon,
+  saveIcon,
+  options,
+  url
```
You don't have to add all these props.
it's not a reusable component, it's a section.
and make the prop name more readable and describes what it used in
the props you need here is just **( userImage, userName, postTime, postCaption, rightImage, leftImage, votesNumber )** and we will add more to it later, but don't handle the styles by the props.

alaasaedy replied:
It's just icons to show every thing as it is in design. Also options prop is coming from the PopUp component a component you made.

MAlngaawy replied:
- these props no need to pass as a props, you need to include it here in the component, it will not change for every post (  divHeight, smallRound, bgColor, pickIcon, searchIcon, saveIcon,)

- **The props just for the values that will change in each post**

- these props need to rewrite to be descriptive for the position it uses in (title, subTitle, imgURL, paragraph, bgImageOne, bgImageTwo, url)

MAlngaawy commented on src/components/PostSection/index.jsx:
```diff
@@ -0,0 +1,117 @@
+          <div>
+            <ReusableDiv>
+              <ShareBtn url={url} />
+            </ReusableDiv>
```
Why do you use the `ReusableDiv` component in the share button and vote and save button? 
please don't overuse the reusable div component if there is no need for that

MAlngaawy reviewed (changes requested):
nice work Alaa, but please consider these changes and the changes in the  comments
-  consider the responsive for the bottom div that contains the (vote and save and share buttons).

- consider the image's width in the different screens especially on the tablet screen

- fix the **OR** circle position at the different screens

- **`reusableDiv` component just use if you have a div the center children, just it**

alaasaedy committed:
- move static data to inside the component and used it.

- restructure the component and use normal tags instead of the usage of reusable div component.

- change props names to more descriptive names.

- add necessary props and passing it to the component.

alaasaedy committed:
- restore Reusable Div component default styling.

- structure Post component to fit different use case inside website.

MAlngaawy committed:
handle some responsoive styles & edit some styles in the small components

MAlngaawy reviewed (approved):
LGTM

MAlngaawy committed:
Merge branch 'develop' of https://github.com/m3ntorship/pickly.io into feature/post-section
